### ISSUE TITLE
losses.softmax_cross_entropy documentation on tensor rank

### ISSUE THREAD

**Issue by omasoud** [tensorflow/tensorflow#16263]
### Documentation Issue
It seems `losses.softmax_cross_entropy()` works just fine with any tensor rank/shape (as long as the last dimension is classes) and not just shapes of `[batch_size, num_classes]` as the documentation indicates. (The documentation also indicates weights should be rank 0 or 1; also not true)

If this is the expected behavior, fixing the documentation would help people avoid doing unnecessary reshapes.

**Comment by aselle** [comment]
@wolffg, could you look at fixing this?

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yongtang** [comment]
Added a PR #17443 for the fix.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @wolffg: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
